### PR TITLE
add env_logger (fix #3869)

### LINKED ISSUE #3869: adopt env_logger
It's very difficult to get debugger log output during the rust unit tests with our current setup.

### PR DESCRIPTION
**Pull request by tokiedokie**
#3869

using env_logger for logging

### CODE REVIEW

**Comment by tokiedokie** [comment]
some tests won't be passed because of #7185

**Comment by bartlomieju** [comment]
@tokiedokie we still want messages like `Check <file>` or `Download <file>` to go to stderr - can `env_logger` be configured in such a way to debug messages go to stdout, while warnings and errors got to stderr?

**Comment by bartlomieju** [comment]
@tokiedokie one more thing; while your working on the logger, could you try to address https://github.com/denoland/deno/issues/6641 as well?

**Comment by tokiedokie** [comment]
@bartlomieju thank you for your feedback
I couldn't find the way to configure env_logger in such a way to debug messages go to stdout, while warnings and errors got to stderr
I keep making all logs go to stderr

I will try to find #6641

**Comment by tokiedokie** [comment]
@bartlomieju I confirmed warn log (#6641)
Should I change default log level to Error so deno doesn't output warn log?

**Comment by bartlomieju** [comment]
> @bartlomieju I confirmed warn logs (#6641)
> Should I change default leg level from Info to Error so deno doesn't output warn logs by default?

@tokiedokie I think that might brake other output after all... Let's skip this issue in this PR. Could you rebase your branch so we can land it?

**Comment by tokiedokie** [comment]
@bartlomieju @ry Done. Please review

File: cli/tests/debug_with_log_level_and_RUST_LOG.out [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Types: [WILDCARD]
+Instantiations: [WILDCARD]
+Parse time: [WILDCARD]
+Bind time: [WILDCARD]
+Check time: [WILDCARD]
+Emit time: [WILDCARD]
+Total TS time: [WILDCARD]
+Compile time: [WILDCARD]
+compiled filename: "[WILDCARD]002_hello.ts.js"
+register_complete file://[WILDCARD]/cli/tests/002_hello.ts
```
Comment (ry) [inline cli/tests/debug_with_log_level_and_RUST_LOG.out:25]: These tests are too specific. We'll have to update the every time we change a debug output. We just want something that checks the logging produces some output - not a specific output. 

I will fix this up. Thanks for the good start @tokiedokie

**Review by ry — APPROVED** [review]
LGTM - thank you @tokiedokie

**Pull request merged**